elastic/kibana
# [Upgrade Assistant] Refactor telemetry

Fixes https://github.com/elastic/kibana/issues/82176

Changes in this PR include:
- Removing a number of items being tracked by the custom usage collector, in favor of [UI counters](https://github.com/elastic/kibana/blob/master/src/plugins/usage_collection/README.mdx#ui-counters) on the client. The custom usage collector is now only responsible for tracking whether deprecation logging is enabled or not. (https://github.com/elastic/kibana/pull/112177/commits/5ec2572099930976806fd066ac8ceb42abb0fd5a)
- Adding several new UI counters to UA (https://github.com/elastic/kibana/pull/112177/commits/f483ab79d229551239231fe4da65c35558165cf5)

**Telemetry coverage**

Overview page
- Component loaded
- Click event for "Create snapshot" button
- Click event for "View deprecation logs in Observability" link
- Click event for "Analyze logs in Discover" link
- Click event for "Reset counter" button

ES deprecations page
- Component loaded
- Click events for starting and stopping reindex tasks
- Click events for upgrading or deleting a Machine Learning snapshot
- Click event for deleting a deprecated index setting

Kibana deprecations page
- Component loaded
- Click event for "Quick resolve" button

**How to test**
1. Navigate to the different pages in UA, click buttons/links as outlined in the section above.
2. If you have your browser's dev tools open, you should see logging similar to:
```
loaded Metric -> (upgrade_assistant:overview_page_load):
{type: 'loaded', appName: 'upgrade_assistant', eventName: 'overview_page_load', count: 1}
```
3. You can also call `GET /api/stats?extended=true&legacy=true` and look for the `ui_counters` JSON for `upgrade_assistant` appName.

**How to review**
- ✅ Verify the approach of moving items out of the custom usage collector to UI counters is appropriate (Kibana core team)
- Assess current telemetry coverage and whether there is anything else we would like to track (Stack Management team/feedback from stakeholders)
